de.fhdo.lemma.servicedsl.extractor/src/de/fhdo/lemma/servicedsl/extractor/ServiceDslExtractor.xtend

```diff
@@ -0,0 +1,272 @@
+            default: null
+        }
+        // If it is a technology add it to the names list for later use in the generation
+        if (importTypeKeyword == "technology") 
+        	IMPORTED_TECH_ALIASES.add(^import.name)
+
+        return '''import «importTypeKeyword» from "«^import.importURI»" as «^import.name»'''
+    }
+
+    /**
+     * Extract Microservice
+     */
+    private def generate(Microservice service) { 
+        val preamble = '''«service.visibility.generate» «service.type.generate»'''
+        return '''
+        «service.generateTechAnnotation»
+        «preamble» microservice «service.lemmaName» {
+            «IF service.interfaces.exists[!operations.empty]»
+                «FOR iface : service.interfaces»
+                    «iface.generate»
+                «ENDFOR»
+            «ELSE»
+                [DEFINE_OPERATIONS]
+            «ENDIF»
+        }'''
+    }
+
+    /**
+     * Extracts the name of a Microservice
+     */
+    private def lemmaName(Microservice service) {
+        return if (service.name.matches(QUALIFIED_NAME_WITH_AT_LEAST_ONE_LEVEL_PATTERN))
+                service.name
+            else
+                '''ADD_QUALIFYING_PART.«service.name»'''
+    }
+
+    /**
+     * Extract Visibility of a Microservice
+     */
+    private def generate(Visibility visibility) {
+        return switch(visibility) {
+            case INTERNAL: 'internal'
+            case PUBLIC: 'public'
+            default: 'architecture'
+        }
+    }
+    
+    /**
+     * Extract MicroserviceType of a Microservice
+     */
+    private def generate(MicroserviceType type) {
+        return switch(type) {
+            case INFRASTRUCTURE: 'infrastructure'
+            case UTILITY: 'utility'
+            default: 'functional'
+        }
+    }
+
+    /**
+     * Extract Interface
+     */
+    private def generate(Interface iface) {
+        return '''
+        interface «iface.name» {
+            «FOR o: iface.operations»
+                «o.generate»
+            «ENDFOR»
+        }'''
+    }
+
+    /**
+     * Extract Endpoint
+     */
+    private def generate(Endpoint endpoint) {
+    	// Formatting is kinda ugly because otherwise xtend's string 
+    	// templates have additional not intended linebreaks
+        return '''«FOR ep: endpoint.protocols SEPARATOR '; '»
+        	«ep.importedProtocol.generate»:«
+        ENDFOR
+        »«FOR ea: endpoint.addresses SEPARATOR ', '
+        	»"«ea»"«
+        ENDFOR»;'''
+    }
+
+    /**
+     * Extract Operation
+     */
+    private def generate(Operation operation) {
+        val endpoints ='''
+        @endpoints(«FOR e: operation.endpoints»«e.generate»«ENDFOR»)
+        '''
+        
+        val aspects = '''
+        «FOR  a: operation.aspects»«a.generate»«ENDFOR»
+        '''
+        
+        val parameters = String.join(", ", operation.parameters.map[generate])
+
+        val comment = if (operation.apiOperationComment !== null) {
```

Comment:
Move before determining the value of the `endpoints` variable to be consistent with the usage of the variables in line 171.